src/bin/cargo/commands/add.rs

```diff
@@ -193,6 +194,12 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     };
     add(&ws, &options)?;
 
+    if !dry_run {
+        // Reload the workspace since we've changed dependencies
+        let ws = args.workspace(config)?;
+        generate_lockfile(&ws)?;
```

Comment:
My original intent is that this will only do the minimal changes necessary to update the lockfile. This function will update every dependency (similar to a blanket `cargo update`). This could potentially move unrelated dependency versions past what the user wanted.

I think using [`resolve_ws`](https://github.com/rust-lang/cargo/blob/85b500ccad8cd0b63995fd94a03ddd4b83f7905b/src/cargo/ops/resolve.rs#L63) will accomplish what I was thinking.

Otherwise, can you say more if this was chosen deliberately?

Reply:
Misunderstood.  I had assume `cargo generate-lockfile` did not do an update.

Reply:
Updated